examples/admin_create_topic/admin_create_topic.go

```diff
@@ -0,0 +1,93 @@
+// Create topic
+package main
+
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"context"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+
+	if len(os.Args) != 5 {
+		fmt.Fprintf(os.Stderr,
+			"Usage: %s <broker> <topic> <partition-count> <replication-factor>\n",
+			os.Args[0])
+		os.Exit(1)
+	}
+
+	broker := os.Args[1]
+	topic := os.Args[2]
+	numParts, err := strconv.Atoi(os.Args[3])
+	if err != nil {
+		fmt.Printf("Invalid partition count: %s: %v\n", os.Args[3], err)
+		os.Exit(1)
+	}
+	replicationFactor, err := strconv.Atoi(os.Args[4])
+	if err != nil {
+		fmt.Printf("Invalid replication factor: %s: %v\n", os.Args[4], err)
+		os.Exit(1)
+	}
+
+	// Create a new AdminClient.
+	// AdminClient can also be instantiated using an existing
+	// Producer or Consumer instance, see NewAdminClientFromProducer and
+	// NewAdminClientFromConsumer.
+	a, err := kafka.NewAdminClient(&kafka.ConfigMap{"bootstrap.servers": broker})
+	if err != nil {
+		fmt.Printf("Failed to create Admin client: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Contexts are used to abort or limit the amount of time
+	// the Admin call blocks waiting for a result.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create topics on cluster.
+	// Set Admin options to wait for the operation to finish (or at most 60s)
+	maxDur, err := time.ParseDuration("60s")
+	if err != nil {
+		panic("ParseDuration(60s)")
+	}
+	results, err := a.CreateTopics(
+		ctx,
+		// Multiple topics can be created simultaneously
+		// by providing more NewTopic structs here.
+		[]kafka.NewTopic{{
+			Topic:             topic,
+			NumPartitions:     numParts,
+			ReplicationFactor: replicationFactor}},
+		// Admin options
+		kafka.SetAdminOperationTimeout(maxDur))
```

Comment:
why not just keep it simple and just use a plain old `CreateTopicsOptions` struct for this? the way this works atm is intricate enough that I need to go investigate `SetAdminOperationTimeout` to understand what it means to be 'setting' something in the context of a method argument. I don't think there's any benefit to trying to have fancy modeling of the the options structure, but there is benefit to having it straightforward.

Reply:
i should say that it is an interesting and quite nice solution you've come up with here, which works effectively, but if it's not familiar to people using the api, the extra complexity is a detractor. if it's idiomatic go, you can ignore the above comment.

Reply:
The problem with using options structs is that uninitialized fields will be set to their "zero value", which is 0 for most data types, but 0 might not actually be the default value of certain options and may have specific meaning.
One way around that is to use a factory function, but since there are no default-args for Go functions those factory functions would need to take all options as arguments and then we're back to the same problem with zero value.
Another way is to use pointers, but that will require temporary variables, allocations or a set of helper functions to provide int/bool/whatnot arguments in pointer form, and that just becomes messy.

I discussed this with @travisjeffery and he suggested the solution that is used here, which I think also reads quite nice.

As for UX:  this solution should actually be easier on the user since if they dont want options (which most people probably wont start with) then they don't have to care about options at all but simply omit that argument.

Reply:
thanks for the commentary - yes, this looks good to me. I'm increasingly of the opinion that golang makes the wrong tradeoffs, but that's a separate issue.